core/trino-main/src/main/java/io/trino/execution/QueryInfo.java

```diff
@@ -50,7 +50,6 @@
     private final QueryId queryId;
     private final SessionRepresentation session;
     private final QueryState state;
-    private final boolean scheduled;
```

Comment:
I wonder if `scheduled` should be kept in QueryInfo and removed from QueryStats as it seems to be about a query's state rather than statistics.
Will let @sopel39 decide this

Reply:
Yeah, I wasn’t totally sure and it’s easy enough to keep the field in both places (just redundant)

Reply:
@sopel39 - think you'll have a chance to weigh in on this soon?

Reply:
> I wonder if scheduled should be kept in QueryInfo and removed from QueryStats as it seems to be about a query's state rather than statistics.

makes sense.

BTW: do we need `scheduled` since we have `state` in query info?

Reply:
@raunaqmorarka - I took a look at moving `scheduled` from `QueryStats` to `QueryInfo` but unfortunately it's used from within `QueryStats#getProgressPercentage` which makes extracting it a harder problem, so I propose we leave it in `QueryStats` since all `QueryInfo` instances have a `QueryStats` value they can consult.

> BTW: do we need scheduled since we have state in query info?

@sopel39 : We don't "need" it in the sense that you could re-compute it, but the `QueryInfo` level notion of being `scheduled` is not a function of the `QueryState` but rather is set to true when of the query's stages are either done, pending, or running- so it does have semantic meaning on it's own.